GameServerLib/Content/ZipPackage.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GameServerCore.Content;
+using GameServerCore.Domain;
+using log4net;
+using LeagueSandbox.GameServer.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace LeagueSandbox.GameServer.Content
+{
+    public class ZipPackage : IPackage
+    {
+        public string PackageName { get; private set; }
+        public string PackagePath { get; private set; }
+
+        private readonly Dictionary<string, ISpellData> _spellData = new Dictionary<string, ISpellData>();
+        private readonly Dictionary<string, ICharData> _charData = new Dictionary<string, ICharData>();
+
+        private readonly Game _game;
+        private readonly ILog _logger;
+
+        private readonly Dictionary<string, Dictionary<string, List<string>>> _content = new Dictionary<string, Dictionary<string, List<string>>>();
+
+        private static readonly string[] ContentTypes = {
+            "Champions",
+            "Items",
+            "Buffs",
+            "Maps",
+            "Spells",
+            "Stats"
+        };
+
+        public ZipPackage(string zipPackagePath, Game game)
+        {
+            PackagePath = zipPackagePath;
+
+            _game = game;
+            _logger = LoggerProvider.GetLogger();
+        }
+
+        public void LoadPackage(string packageName)
+        {
+            PackageName = packageName;
+
+            InitializeContent();
+            LoadPackage();
+            LoadItems();
+            LoadScripts();
+        }
+
+        private void InitializeContent()
```

Comment:
These common functions to all packages should be in some abstract class `PackageBase`.